pandas/core/indexes/base.py

```diff
@@ -395,10 +395,10 @@ def __new__(
                 raise ValueError("Index data must be 1-dimensional")
             return cls._simple_new(subarr, name)
 
-        elif hasattr(data, "__array__"):
-            return Index(np.asarray(data), dtype=dtype, copy=copy, name=name, **kwargs)
         elif data is None or is_scalar(data):
             raise cls._scalar_data_error(data)
+        elif hasattr(data, "__array__"):
+            return Index(np.asarray(data), dtype=dtype, copy=copy, name=name, **kwargs)
```

Comment:
Just checking for scalars _before_ testing for an `__array__` attribute since things like `np.array([1, 2])[0]` pass both